graphbolt/src/cuda/index_select_impl.cu

```diff
@@ -14,6 +15,68 @@
 namespace graphbolt {
 namespace ops {
 
+static bool GetEnv(const std::string& env) {
+  // Get the environment variable.
+  // If the environment variable is not set or set to 0, return false.
+  // Otherwise, return true.
+  const char* val = std::getenv(env.c_str());
+  if (val == nullptr) {
+    return false;
+  }
+  if (strcmp(val, "0") == 0) {
+    return false;
+  }
+  return true;
+}
+
+template <typename T>
+int NumberOfBits(const T& range) {
+  if (range <= 1) {
+    // ranges of 0 or 1 require no bits to store
+    return 0;
+  }
+
+  int bits = 1;
+  while (bits < static_cast<int>(sizeof(T) * 8) && (1 << bits) < range) {
```

Comment:
There was a minor bug which I fixed in #6586. It might be relevant when testing 3.5B int8 tensor test case.

Reply:
But since we are testing up to 4B, the current code may still work though.